components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1931,8 +1931,11 @@ private AccessTokenExtendedAttributes getExtendedTokenAttributes(OAuthMessage oA
         try {
             ServiceProvider serviceProvider = getServiceProvider(oauth2Params.getClientId());
             // TODO: Improve to read the script separately instead of reading from adaptive script.
-            if (!EndpointUtil.isExternalConsentPageEnabledForSP(serviceProvider) ||
-                    serviceProvider.getLocalAndOutBoundAuthenticationConfig().getAuthenticationScriptConfig() == null) {
+            AuthenticationScriptConfig scriptConfig = serviceProvider.getLocalAndOutBoundAuthenticationConfig()
+                    .getAuthenticationScriptConfig();
+            boolean isValidScript = scriptConfig != null && scriptConfig.getContent() != null &&
+                    scriptConfig.getContent().contains(DYNAMIC_TOKEN_DATA_FUNCTION);
+            if (!isValidScript) {
```

Comment:
ok I see you have moved this to the invocation point. Can we have a separate function like "implementsTokenAttributeExtension (ServiceProvider sp)" which checks this logic and return true or false and then allows invoking this "getExtendedTokenAttributes" function

Reply:
Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2437/commits/6ec88a197a4a6608bbc8a1e69f6c53c085d4f54f